lib/liveman/survey/surveys.ex

```diff
@@ -8,19 +8,56 @@ defmodule Liveman.Survey.Surveys do
   def list_surveys do
     survey_list_json = fecth_surveys_from_file!()
 
-    Enum.map(survey_list_json, fn survey_json ->
-      build_survey(survey_json)
-    end)
+    get_survey_list(survey_list_json)
   end
 
   defp fecth_surveys_from_file! do
     {_, body} = File.read(@json_file)
     Jason.decode!(body)
   end
 
+  def create_answers(params) do
+    survey = get_survey(params["survey_id"])
+
+    validate(survey, params["questions"])
+  end
+
+  defp validate(nil, _questions), do: {:error, :invalid_survey}
+  defp validate(_survey, nil), do: {:error, :invalid_questions}
+
+  defp validate(survey, questions) do
+    survey_questions_ids = get_survey_question_ids(survey.questions)
+    question_ids = get_question_ids(questions)
+
+    # Check if all submitted questions are belong to the survey
+    if question_ids -- survey_questions_ids == [] do
+      :ok
+    else
+      {:error, :invalid_questions}
+    end
+  end
+
+  defp get_survey_list(survey_list_json) do
+    Enum.map(survey_list_json, fn survey_json ->
+      build_survey(survey_json)
+    end)
+  end
+
   defp build_survey(json) do
     %Survey{}
     |> Survey.changeset(json)
     |> Ecto.Changeset.apply_changes()
   end
+
+  defp get_survey(survey_id) do
+    Enum.find(list_surveys(), fn survey -> survey_id == survey.id end)
+  end
+
+  defp get_question_ids(questions) do
+    Enum.map(questions, fn question -> question["id"] end)
```

Comment:
Just curious, Is the survey question in `Liveman.Survey.Schemas.Question` schema? We're calling `"id"` instead of `.id` here? 🤔

Reply:
There are actually map which I get from the clients via API, That's why I need to use `"id"` instead of `.id`